src/rust/ide/lib/view/src/graph_editor/component/node/port/output.rs

```diff
@@ -582,43 +611,88 @@ impl OutputPortsData {
 #[derive(Debug,Clone,CloneRef)]
 pub struct OutputPorts {
     /// The FRP api of the `OutPutPorts`.
-    pub frp     : Frp,
-        network : frp::Network,
-        data    : Rc<OutputPortsData>,
+    pub frp              : Frp,
+        network          : frp::Network,
+        network_internal : Rc<RefCell<frp::Network>>,
+        data             : Rc<OutputPortsData>,
+        type_map         : TypeColorMap,
+        expression       : Rc<RefCell<Expression>>,
```

Comment:
Why output ports keep expression information? This is not their responsibility. Is this information duplicated now? If so, it's bad, as we should not duplicate information across strucutres.

Reply:
We are using this here so we can look up the `ast::Id` for a given `&[span_tree::Crumb]`, which we need to determine the color of our ports. See also https://github.com/enso-org/ide/pull/663#discussion_r459478405
This is similar to what happens in the `port::Manager`. Maybe we should unify that?

Reply:
As discussed on discord, lets rename it to `expression` and `pattern` in the appropriate places.